kasa/discover.py

```diff
@@ -63,20 +65,30 @@ def do_discover(self) -> None:
         req = json.dumps(Discover.DISCOVERY_QUERY)
         _LOGGER.debug("[DISCOVERY] %s >> %s", self.target, Discover.DISCOVERY_QUERY)
         encrypted_req = self.protocol.encrypt(req)
+        new_req = binascii.unhexlify('020000010000000000000000463cb5d3')
+#        for i in range(self.discovery_packets):
+#            self.transport.sendto(encrypted_req[4:], self.target)  # type: ignore
+        _LOGGER.debug("[NEW DISCOVERY] %s >> magic_packet", self.target)
         for i in range(self.discovery_packets):
-            self.transport.sendto(encrypted_req[4:], self.target)  # type: ignore
+            self.transport.sendto(new_req, self.new_target)
 
     def datagram_received(self, data, addr) -> None:
         """Handle discovery responses."""
         ip, port = addr
         if ip in self.discovered_devices:
             return
 
-        info = json.loads(self.protocol.decrypt(data))
+        if port == 9999:
```

Comment:
I would suggest that instead of relying on a hardcoded ports, to either have a command line option, or even better - attempt the old encryption mechanism, and if that fails - to try the new one.

In respect to this I've already opened a PR #109 that would extract the port and make it configurable and I think it'd quite handy for this implementation.

Reply:
An example discovery payload is:

```
{'result': 
    {'ip': '<ip-address>',
      'mac': '<mac-address>', 
      'device_id': '<16 octet hex>', 
      'owner': '<16 octet hex>',
      'device_type': 'IOT.SMARTPLUGSWITCH', 
      'device_model': 'HS110(UK)', 
      'hw_ver': '4.1', 
      'factory_default': True, 
      'mgt_encrypt_schm': 
        {'is_support_https': False, 
         'encrypt_type': 'KLAP', 
         'http_port': 80}
    }, 
    'error_code': 0
}
```

So we could use the device_model as a way of determining the different between a plug (HS100 or HS110) and a strip (KP303)